Hardware/HardwareTests/CardReaderTests.swift

```diff
@@ -0,0 +1,61 @@
+import XCTest
+@testable import Hardware
+
+/// Tests the mapping between CardReader and SCPReader
+final class CardReaderTests: XCTestCase {
+    func test_card_reader_maps_serial_number() {
+        let mockReader = MockStripeCardReader.bbpos()
+        let cardReader = CardReader(reader: mockReader)
+
+        XCTAssertEqual(cardReader.serial, mockReader.serialNumber)
```

Comment:
nit: would be nice to have the Given/When/Then annotations so that it's easier to follow the scenario, especially for more complicated test cases with async results 🙂 (same suggestion for the other test cases)

```suggestion
        // Given
        let mockReader = MockStripeCardReader.bbpos()
        
        // when
        let cardReader = CardReader(reader: mockReader)

        // Then
        XCTAssertEqual(cardReader.serial, mockReader.serialNumber)
```